addons/mail/tests/discuss/test_discuss_binary_controller.py

```diff
@@ -0,0 +1,134 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import odoo
+from odoo.addons.mail.tests.test_binary_controller import TestBinaryControllerCommon
+
+
+@odoo.tests.tagged("-at_install", "post_install")
+class TestDiscussBinaryControllerCommon(TestBinaryControllerCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.group = cls.env["discuss.channel"].create({"name": "Group", "channel_type": "group"})
+
+    def _execute_options(self, options, data_user, record):
+        if options.get("with_channel"):
+            self._add_member([data_user, record])
```

Comment:
If I'm reading this code correctly, adding member will actually be persisted across subtests (it's not in a transaction and not cleaned) so we might as well call it explicitly inside each test at the start, rather than letting each sub test do it again.